prov/efa/src/efa_mr.c

```diff
@@ -784,12 +784,17 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, const void *at
 	/* To support Emulated RMA path, if the access is not supported
 	 * by EFA, modify it to FI_SEND | FI_RECV
 	 */
-	core_access = mr_attr.access;
-	if (!core_access || (core_access & ~EFA_MR_SUPPORTED_PERMISSIONS))
-		core_access = FI_SEND | FI_RECV;
+	ofi_access = mr_attr.access;
+	if (!ofi_access || (ofi_access & ~EFA_MR_SUPPORTED_PERMISSIONS))
+		ofi_access = FI_SEND | FI_RECV;
 
-	/* Local read access to an MR is enabled by default in verbs */
-	if (core_access & FI_RECV)
+	/* Local read access to an MR is enabled by default in verbs
+	 *
+	 * We need IBV_ACCESS_LOCAL_WRITE for two emulated cases
+	 * 1. When emulating fi_send with RMA read - ofi_access is set to FI_RECV
+	 * 2. When emulating fi_write with RMA read - ofi_access is set to FI_REMOTE_WRITE
+	 */
+	if (ofi_access & (FI_RECV | FI_REMOTE_WRITE))
```

Comment:
According to https://www.rdmamojo.com/2012/09/07/ibv_reg_mr/, `IBV_ACCESS_LOCAL_WRITE` means 

>Enable Local Write Access: Memory Region can be used in Receive Requests or in IBV_WR_ATOMIC_CMP_AND_SWP or
IBV_WR_ATOMIC_FETCH_AND_ADD to write locally the remote content value

Does it exactly fit our model here that the mr is used to get data via rdma read?

I know we have been do that even before your PR, that an `FI_RECV` access can also involve rdma read in read protocol. I am just curious

Reply:
> Does it exactly fit our model here that the mr is used to get data via rdma read?

I think so. I think the destination buffer of a RDMA read needs to have IBV_ACCESS_LOCAL_WRITE